files/var/www/cgi-bin/j/heartbeat.cgi

```diff
@@ -10,18 +10,26 @@ else
 	soc_temp=""
 fi
 
-case "$vendor" in
-	ingenic)
-		daynight_value=$(imp-control.sh gettotalgain)
-		;;
-	sigmastar)
-		echo 2 >/sys/devices/virtual/mstar/sar/channel
-		daynight_value=$(cat /sys/devices/virtual/mstar/sar/value)
-		;;
-	*)
-		daynight_value=-1
-		;;
-esac
+if [ ! -z $(pidof majestic) ]; then
+	daynight_value=$(curl -s http://127.0.0.1/metrics?value=isp_again)
+fi
+
+if [ -z "$daynight_value" ]; then
+	case "$vendor" in
+		ingenic)
+			daynight_value=$(imp-control.sh gettotalgain)
+			;;
+		sigmastar)
+			if [ -e /sys/devices/virtual/mstar/sar/channel ]; then
```

Comment:
this `if` requres `else` with `daynight_value=-1`, otherwise sigmastar devices with no sar file get no value assigned.

Reply:
This has been addressed with the latest change.
The new changes for the metrics should be available now for testing.